models/repo_collaboration.go

```diff
@@ -186,9 +186,49 @@ func (repo *Repository) DeleteCollaboration(uid int64) (err error) {
 		return err
 	}
 
+	// Unassign a user from any issue (s)he has been assigned to in the
+	// repository
+	if err := repo.removeIssueAssignees(sess, uid); err != nil {
+		return err
+	}
+
 	return sess.Commit()
 }
 
+func (repo *Repository) removeIssueAssignees(e Engine, userID int64) error {
+
+	user, err := GetUserByID(userID)
+	if err != nil {
+		return err
+	}
+
+	perm, err := getUserRepoPermission(e, repo, user)
+	if err != nil {
+		return err
+	}
+
+	if perm.CanWrite(UnitTypeIssues) {
+		return nil
+	}
+
+	if canBeAssigned, err := CanBeAssigned(user, repo, true); canBeAssigned && err == nil {
+		return nil
+	}
+
+	assignee := &IssueAssignees{
+		AssigneeID: userID,
+	}
+
+	_, err = e.
+		Join("INNER", "issue", "`issue`.id = `issues_assignees`.issue_id AND `issue`.repo_id = ?", repo.ID).
```

Comment:
You need to do this twice: once for issues and once for pull requests. You cannot generalize "they will all be PRs" because issues can be assigned too.

Reply:
@guillep2k dont think this is a problem - because PRs are alos represented as issues for assignment

Reply:
(PRs are issues with extend information :) )